experimental/Schemes/AffineSchemes.jl

```diff
@@ -298,64 +460,177 @@ function is_closed_embedding(X::Spec, Y::Spec)
+    X::AbsSpec{BRT, <:MPolyRing},
+    Y::AbsSpec{BRT, <:MPolyQuo}
+  ) where {BRT}
+  R = OO(X)
+  R == base_ring(OO(Y)) || error("schemes can not be compared")
+  return Spec(Y)
+end
+
+function Base.intersect(
+    X::AbsSpec{BRT, <:MPolyRing},
+    Y::AbsSpec{BRT, <:MPolyLocalizedRing}
+  ) where {BRT}
+  R = OO(X)
+  R == base_ring(OO(Y)) || error("schemes can not be compared")
+  return Spec(Y)
+end
+
+function Base.intersect(
+    X::AbsSpec{BRT, <:MPolyRing},
+    Y::AbsSpec{BRT, <:MPolyQuoLocalizedRing}
+  ) where {BRT}
+  R = OO(X)
+  R == base_ring(OO(Y)) || error("schemes can not be compared")
+  return Spec(Y)
+end
+
+function Base.intersect(
+    Y::AbsSpec{BRT, <:Any},
+    X::AbsSpec{BRT, <:MPolyRing}
+  ) where {BRT}
+  return intersect(X, Y)
+end
+
+### For Specs of MPolyQuos
+function Base.intersect(
+    X::AbsSpec{BRT, <:MPolyQuo},
+    Y::AbsSpec{BRT, <:MPolyQuo}
+  ) where {BRT}
   R = base_ring(OO(X))
-  return Spec(R, modulus(OO(X)) + modulus(OO(Y)), inverted_set(OO(X))*inverted_set(OO(Y)))
+  R == base_ring(OO(Y)) || error("schemes can not be compared")
+  return Spec(quo(R, modulus(OO(X)) + modulus(OO(Y)))[1])
 end
 
 function Base.intersect(
-    X::Spec{BRT, BRET, RT, RET, MST1}, 
-    Y::Spec{BRT, BRET, RT, RET, MST2}
-  ) where {BRT, BRET, RT, RET, MST1<:MPolyPowersOfElement{BRT, BRET, RT, RET}, MST2<:MPolyPowersOfElement{BRT, BRET, RT, RET}}
-  base_ring(OO(X)) == base_ring(OO(Y)) || error("schemes can not be intersected")
-  issubset(X, Y) && return X
-  issubset(Y, X) && return Y
-  UX = inverted_set(OO(X))
-  UY = inverted_set(OO(Y))
-  U = UX*UY
-  IX = modulus(OO(X))
-  IY = modulus(OO(Y))
-  I = IX + IY
+    X::AbsSpec{BRT, <:MPolyQuo},
+    Y::AbsSpec{BRT, <:MPolyLocalizedRing}
+  ) where {BRT}
+  R = base_ring(OO(X))
+  R == base_ring(OO(Y)) || error("schemes can not be compared")
+  return Spec(quo(OO(Y), OO(Y)(modulus(OO(X))))[1])
+end
+
+function Base.intersect(
+    X::AbsSpec{BRT, <:MPolyQuo},
+    Y::AbsSpec{BRT, <:MPolyQuoLocalizedRing}
+  ) where {BRT}
+  R = base_ring(OO(X))
+  R == base_ring(OO(Y)) || error("schemes can not be compared")
+  return Spec(quo(OO(Y), OO(Y)(modulus(OO(X))))[1])
+end
+
+function Base.intersect(
+    Y::AbsSpec{BRT, <:Any},
+    X::AbsSpec{BRT, <:MPolyQuo}
+  ) where {BRT}
+  return intersect(X, Y)
+end
+
+### For Specs of MPolyLocalizedRings
+function Base.intersect(
+    X::AbsSpec{BRT, <:MPolyLocalizedRing},
+    Y::AbsSpec{BRT, <:MPolyLocalizedRing}
+  ) where {BRT}
+  R = base_ring(OO(X))
+  R == base_ring(OO(Y)) || error("schemes can not be compared")
+  return Spec(Localization(R, inverted_set(OO(X)) * inverted_set(OO(Y)))[1])
+end
+
+function Base.intersect(
+    X::AbsSpec{BRT, <:MPolyLocalizedRing},
+    Y::AbsSpec{BRT, <:MPolyQuoLocalizedRing}
+  ) where {BRT}
+  R = base_ring(OO(X))
+  R == base_ring(OO(Y)) || error("schemes can not be compared")
+  return Spec(quo(OO(Y), OO(Y)(modulus(OO(X))))[1])
```

Comment:
stupid question: don't you need to pay attention to the multiplicatively closed subsets used in the localizations of the two schemes (just as in the preceding instance)?

Reply:
Oops. Yes, I was missing out on that.

Reply:
Now that I'm looking at it again: No, only one of the rings has a multiplicative set and it's preserved, because I'm calling `quo` on `OO(Y)`.